internal/quickfort/dig.lua

```diff
@@ -457,52 +457,115 @@ local function do_traffic_restricted(ctx)
+    i={action=do_up_down_stair, use_priority=true},
+    r={action=do_ramp, use_priority=true},
+    z={action=do_remove_ramps, use_priority=true},
+    t={action=do_mine, use_priority=true},
+    p={action=do_gather, use_priority=true},
+    s={action=do_smooth, use_priority=true},
+    e={action=do_engrave, use_priority=true},
+    F={action=do_fortification, use_priority=true},
+    T={action=do_track, use_priority=true},
+    v={action=do_toggle_engravings},
+    -- the semantics are unclear if the code is M but m or force_marker_mode is
+    -- also specified. skipping all other marker mode settings when toggling
+    -- marker mode seems to make the most sense.
+    M={action=do_toggle_marker, skip_marker_mode=true},
+    n={action=do_remove_construction, use_priority=true},
+    x={action=do_remove_designation},
+    bc={action=do_claim},
+    bf={action=do_forbid},
+    bm={action=do_melt},
+    bM={action=do_remove_melt},
+    bd={action=do_dump},
+    bD={action=do_remove_dump},
+    bh={action=do_hide},
+    bH={action=do_unhide},
+    oh={action=do_traffic_high},
+    on={action=do_traffic_normal},
+    ol={action=do_traffic_low},
+    ['or']={action=do_traffic_restricted},
 }
 
-local function dig_tile(ctx, code, marker_mode)
+-- handles marker mode 'm' prefix and priority suffix
+local function extended_parser(_, keys)
+    local marker_mode = false
+    if keys:startswith('m') then
+        keys = string.sub(keys, 2)
+        marker_mode = true
+    end
+    local found, _, code, priority = keys:find('^(%D*)(%d*)$')
+    if not found then return nil end
+    if #priority == 0 then
+        priority = 4
+    else
+        priority = tonumber(priority)
+        if priority < 1 or priority > 7 then
+            log('priority must be between 1 and 7 (inclusive)')
+            return nil
+        end
+    end
+    if #code == 0 then code = 'd' end
+    if not rawget(dig_db, code) then return nil end
+    local custom_designate = copyall(dig_db[code])
+    custom_designate.marker_mode = marker_mode
+    custom_designate.priority = priority
+    return custom_designate
+end
+
+setmetatable(dig_db, {__index=extended_parser})
+
+local function get_priority_block_square_event(block_events)
+    for i,v in ipairs(block_events) do
+        if v:getType() == df.block_square_event_type.designation_priority then
+            return v
+        end
+    end
+    return nil
+end
+
+-- modifies any existing priority block_square_event to the specified priority.
+-- if the block_square_event doesn't already exist, create it iff we're setting
+-- the priority to any number but 4 (the default)
+local function set_priority(ctx, priority)
+    log('setting priority to %d', priority)
+    local block_events = dfhack.maps.getTileBlock(ctx.pos).block_events
+    local pbse = get_priority_block_square_event(block_events)
+    if not pbse then
+        if priority == 4 then return end
+        print("creating event")
```

Comment:
Probably should be a `log()` call

Reply:
it was a leftover dev debugging line; removed.